### PR TITLE
Add unit tests for Microsoft.Testing.Platform

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Platform/Microsoft.Testing.Platform/Logging/FileLogger.cs [hunk 1]
```diff
@@ -102,11 +100,13 @@ public FileLogger(string logFolder, string fileName, IClock clock, LogLevel logL
     private static FileStream OpenFileStreamForAppend(string fileName)
         => new(fileName, FileMode.Append, FileAccess.Write, FileShare.Read);
 
-    private static FileStream CreateFileStream(string fileName)
-        => new(fileName, FileMode.CreateNew, FileAccess.Write, FileShare.Read);
-
-    private FileStream CreateFileStream()
+    private FileStream CreateFileStream(string? fileName)
```
Comment (MarcoRossignoli) [inline src/Platform/Microsoft.Testing.Platform/Logging/FileLogger.cs]: default null and avoid the above null?

File: test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <UseVSTestRunner>false</UseVSTestRunner>
+    <ImplicitUsings>true</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <!-- Testing configuration -->
+  <PropertyGroup>
+    <GenerateTestingPlatformEntryPoint>false</GenerateTestingPlatformEntryPoint>
+    <EnableCodeCoverage>false</EnableCodeCoverage>
+    <DefineConstants Condition=" '$(EnableCodeCoverage)' == 'True' ">$(DefineConstants);ENABLE_CODECOVERAGE</DefineConstants>
```
Comment (MarcoRossignoli) [inline test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj:20]: do we keep code cov constant?

File: test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  <ItemGroup>
+    <None Update="testsbaseline.netfx.txt">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="testsbaseline.txt">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="Microsoft.Testing.Platform.UnitTests.launcher.config.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (MarcoRossignoli) [inline test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj:52]: don't recall if this is needed or we're using the msbuild feature

**Review by MarcoRossignoli — DISMISSED** [review]

File: test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj (hunk 2 above)
Comment (Evangelink) [reply test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj:20]: It's used in code for the registration line and using

File: test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj (hunk 3 above)
Comment (Evangelink) [reply test/UnitTests/Microsoft.Testing.Platform.UnitTests/Microsoft.Testing.Platform.UnitTests.csproj:52]: It's still present on TA so I have it. If you are done with this part and we need to update TA tests, please let me know.

**Review by nohwnd — APPROVED** [review]

**Pull request merged**